File: programs/wc.r
#!/usr/bin/perl -w
use strict;

my $line = <>;
chomp $line;
my ($key, $sum) = split /\t/, $line;

while(<>){
  chomp;
  my($newkey, $val) = split /\t/;
  if($key eq $newkey){
    $sum += $val;
    next;
  }

  print "$key\t$sum\n";
  $key = $newkey;
  $sum = $val;
}

print "$key\t$sum\n" if $key;

File: programs/wc.m
#!/usr/bin/perl -w
use strict;

$/ = $ENV{TTRS} if exists $ENV{TTRS};
$/ = eval $/ if ($/ =~ m|^['"]|);

while(<>){
  my %words;
  $words{$_}++ for split /\W+/;
  delete $words{''} if exists $words{''};
  print "$_\t$words{$_}\n" for keys %words;
  #print STDERR "Got ", scalar keys %words, " unique words in file.\n";
}



